extensions/transfer-functions/transfer-functions-core/src/test/java/org/eclipse/dataspaceconnector/transfer/functions/core/TransferFunctionsCoreHttpTest.java

```diff
@@ -0,0 +1,95 @@
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.transfer.functions.core;
+
+import okhttp3.MediaType;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.eclipse.dataspaceconnector.junit.launcher.EdcExtension;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessManager;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferWaitStrategy;
+import org.eclipse.dataspaceconnector.spi.types.domain.metadata.DataEntry;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+import org.eclipse.dataspaceconnector.transfer.functions.spi.flow.http.TransferFunctionInterceptorRegistry;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static okhttp3.Protocol.HTTP_1_1;
+import static org.eclipse.dataspaceconnector.transfer.functions.core.TransferFunctionsCoreServiceExtension.ENABLED_PROTOCOLS_KEY;
+import static org.eclipse.dataspaceconnector.transfer.functions.core.TransferFunctionsCoreServiceExtension.TRANSFER_TYPE;
+
+/**
+ * Verifies the HTTP flow controller works.
+ */
+@ExtendWith(EdcExtension.class)
+public class TransferFunctionsCoreHttpTest {
+
+    @Test
+    void verifyHttpFlowControllerInvoked(TransferProcessManager processManager, TransferFunctionInterceptorRegistry registry) throws InterruptedException {
+
+        final var latch = new CountDownLatch(1);
+        var invoked = new AtomicBoolean();
+
+        registry.registerHttpInterceptor(chain -> {
+            invoked.set(true);
+            latch.countDown();
+            return new Response.Builder()
+                    .request(chain.request())
+                    .protocol(HTTP_1_1).code(200)
+                    .body(ResponseBody.create("", MediaType.get("application/json"))).message("")
+                    .build();
+        });
+
+        var dataEntry = DataEntry.Builder.newInstance().id("test123").build();
+
+        var dataRequest = DataRequest.Builder.newInstance()
+                .id(UUID.randomUUID().toString())
+                .protocol("ids")
+                .destinationType("foo")
+                .dataEntry(dataEntry)
+                .managedResources(false)
+                .dataDestination(DataAddress.Builder.newInstance().type("test-protocol1").build())
+                .connectorId("test").build();
+
+        processManager.initiateProviderRequest(dataRequest);
+
+        latch.await(1, TimeUnit.MINUTES);
+        Assertions.assertTrue(invoked.get());
```

Comment:
we can always `assertTrue(latch.await(...))`, which would save us the `assertTrue(invoked.get())`. 
```suggestion
       assertThat(latch.await(1, TimeUnit.MINUTES)).isTrue()
```
Also, I think we should try and stick with one assertion framework (here: AssertJ)